schema.json

```diff
@@ -1,255 +1,266 @@
+        "name": { "$ref": "#/definitions/nameType" },
+        "type": {
+          "enum": [
+            "int8",
+            "int16",
+            "int32",
+            "int64",
+            "uint8",
+            "uint16",
+            "uint32",
+            "uint64"
+          ]
         },
-        {
-          "properties": {
-            "type": {
-              "enum": ["ipv4"]
-            },
-            "name": {
-              "$ref": "#/definitions/name"
-            }
-          },
-          "required": ["name", "type"]
+        "labels": {
+          "description": "A list of labels that name special values.",
+          "type": "array",
+          "items": { "$ref": "#/definitions/labeledIntegerType" },
+          "uniqueItems": true
         },
-        {
-          "properties": {
-            "type": {
-              "enum": ["mac"]
-            },
-            "name": {
-              "$ref": "#/definitions/name"
-            }
-          },
-          "required": ["name", "type"]
+        "prefix": {
+          "description": "The unit prefix, defined in Table A-14 of E1.20.",
+          "type": "integer",
+          "minimum": 0,
+          "maximum": 255
         },
-        {
-          "properties": {
-            "type": {
-              "enum": ["uid"]
-            },
-            "name": {
-              "$ref": "#/definitions/name"
-            }
-          },
-          "required": ["name", "type"]
+        "ranges": {
+          "description": "A list of possible ranges for the value. The
+                          complete range is the union of all the ranges and
+                          labels.",
+          "type": "array",
+          "items": { "$ref": "#/definitions/rangeType" },
+          "uniqueItems": true
         },
-        {
-          "properties": {
-            "type": {
-              "enum": ["group"]
-            },
-            "name": {
-              "$ref": "#/definitions/name"
-            },
-            "fields": {
-              "type": "array",
-              "items": {
-                "$ref": "#/definitions/field"
-              }
-            },
-            "max_size": {
-              "type": "integer",
-              "minimum": 0,
-              "exclusiveMinimum": true
-            },
-            "min_size": {
-              "type": "integer",
-              "minimum": 0
-            }
-          },
-          "required": ["name", "fields", "type"]
+        "unit": {
+          "description": "The unit type, defined in Table A-13 of E1.20.",
+          "type": "integer",
+          "minimum": 0,
+          "maximum": 255
         }
-      ]
+      },
+      "required": [ "type" ]
+    },
+    "labeledIntegerType": {
+      "description": "Associates a name to an integer value.",
+      "type": "object",
+      "properties": {
+        "name": { "$ref": "#/definitions/nameType" },
+        "value": { "type": "integer" }
+      },
+      "required": [ "value" ]
```

Comment:
Name should be required too, this dropped out from our last discussion. Otherwise it's just a range with a single value.